File: day1/python/day1.py
nls= list(map(int, open('input/inputday1.txt', 'r').read().strip('\n')))
#nls = [1,2,1,3,1,4,1,5]
size = len(nls)

def part1 (inputlist):
  sm = 0
  inputlist.append(inputlist[0])
  for i in range(0, size):
    if inputlist[i] == inputlist[i+1]:
      sm += inputlist[i]
  return sm


def part2 (inputlist):
  sm = 0
  inputlist.extend(inputlist)
  for i in range(0, size):
    if inputlist[i] == inputlist[i+(size/2)]:
      sm += inputlist[i]
  return sm

#print part1(nls)
#print part2(nls)

File: day1/ruby/day1.rb
ls = File.read("input/inputday1.txt").delete("\n").split('').to_a
#ls = "122456".chars.to_a


def part1(input)
  input << input.first
  return input.each_cons(2).to_a
              .select{|a,b| a==b}
              .map{|a,b| a.to_i}
              .reduce(:+)
end

def part2(input)
  size = input.size
  return input.select
              .each_with_index { |a, i| input[(i + size / 2) % size] == a }
              .map(&:to_i)
              .inject(:+)
end


p part1(ls)
